bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigTomlParser.java

```diff
@@ -286,6 +287,9 @@ private static void validateKeyField(TomlTableNode recordTable, String[] fieldNa
     }
 
     private static Object getBalValue(String variableName, int typeTag, Object tomlValue) {
+        if (typeTag == TypeTags.BYTE_TAG) {
+            return ((Long) tomlValue).intValue();
```

Comment:
What happens if someone specifies config value for byte which is out of range?

Reply:
Can we get classcastexception in here? ex. specify string value for byte

Reply:
Class cast exception will not happen here, since we check the type of value before the cast. 
I'll add a check for the range.